src/test/java/uk/gov/hmcts/darts/datamanagement/service/DataManagementServiceImplTest.java

```diff
@@ -242,4 +250,132 @@ void testCopyDataRethrowsExceptionsAsDartsApiException() {
                 UUID.randomUUID().toString(),
                 UUID.randomUUID().toString()));
     }
+
+
+    @Test
+    void positiveGetChecksumTypical() {
+        final String checksum = "abc123";
+        final byte[] checkSumBytes = checksum.getBytes();
+        final UUID blobId = UUID.fromString("431318c8-97db-415c-b321-120c48f0ffe2");
+        final String containerName = "container123";
+        final String connectionString = "connectionString";
+
+        BlobServiceClient blobServiceClient = mock(BlobServiceClient.class);
+        BlobContainerClient blobContainerClient = mock(BlobContainerClient.class);
+        BlobProperties blobProperties = mock(BlobProperties.class);
+        when(blobProperties.getContentMd5())
+            .thenReturn(checkSumBytes);
+        BlobClient blobClient = mock(BlobClient.class);
+        when(blobClient.getProperties())
+            .thenReturn(blobProperties);
+
+        when(dataManagementFactory.getBlobServiceClient(any()))
+            .thenReturn(blobServiceClient);
+        when(dataManagementFactory.getBlobContainerClient(any(), any()))
+            .thenReturn(blobContainerClient);
+        when(dataManagementFactory.getBlobClient(any(), any()))
+            .thenReturn(blobClient);
+        when(dataManagementConfiguration.getBlobStorageAccountConnectionString())
+            .thenReturn(connectionString);
+        when(fileContentChecksum.encodeToString(checkSumBytes))
+            .thenReturn(checksum);
+
+        when(blobClient.exists()).thenReturn(true);
+
+
+        assertThat(dataManagementService.getChecksum(containerName, blobId))
+            .isEqualTo(checksum);
+
+        verify(dataManagementConfiguration, times(1)).getBlobStorageAccountConnectionString();
+        verify(dataManagementFactory, times(1)).getBlobServiceClient(connectionString);
+        verify(dataManagementFactory, times(1)).getBlobContainerClient(containerName, blobServiceClient);
+        verify(dataManagementFactory, times(1)).getBlobClient(blobContainerClient, blobId);
+        verify(blobClient, times(2)).exists();
+        verify(blobClient, times(1)).getProperties();
+        verify(blobProperties, times(1)).getContentMd5();
+        verify(fileContentChecksum, times(1)).encodeToString(checkSumBytes);
+    }
+
+    @Test
+    void negativeGetChecksumFileNotFoundExistsFalse() {
+        assertGetChecksumNotFound(false);
+    }
+
+    @Test
+    void negativeGetChecksumFileNotFoundExistsNull() {
+        assertGetChecksumNotFound(null);
+    }
+
+    @Test
+    void negativeGetChecksumChecksumNotFound() {
```

Comment:
Checksum in test name twice

Reply:
This is by design getChecksum is the method name checksumNotFound is the what we are testing